app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/CalendarAdapter.java

```diff
@@ -110,8 +112,10 @@ public void refreshCalendarView(Context ctx, DocumentReference currentHouse, Str
                         calendar.setEvents(newEvents);
                         if(currentHouse != null) LocalStorage.pushEventsOffline(ctx, currentHouse.getId(), newEvents);
                         generateItems(newEvents);
+                        EspressoIdlingResource.decrement();
                     } else {
                         logAndToast(ctx.toString(), errMessage, task.getException(), ctx, ctx.getString(R.string.refresh_calendar_fail));
+                        EspressoIdlingResource.decrement();
```

Comment:
you can probably move the two decrements to just one after the if else

Reply:
Good call! Done in last commit